Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Use the base Python 3.6 image.  It is sufficient for accomplishing
+# most of the stuff, and includes some handy build tools like gcc.
+#
+# NOTE This could be changed to something like `python:latest` if we
+# find that there is no dependency on 3.6 itself.
+FROM python:3.6
```

Comment:
(The gcc dep is important because different versions provide different error messages, as well as different features, so while your gcc might say "you probably missed a return" theirs might just say "error at line 41", so it can affect how harshly you grade the effort portion.)